### PR TITLE
dep: pull the deps from maven.

### PR DESCRIPTION
**Pull request by ckipp01**
To better keeps the deps up to date we can actually use the built
in maven feature to pull the latest.

### CODE REVIEW

**Comment by lefou** [comment]
Thanks Chris, I wasn't aware of the feature of giter8. Resolving the latest versions at initialization time is for sure a nice feature. Unfortunately, we loose some control over whether the resolved version combinations are guaranteed to work as expected.

What I would prefer is direct control over versions which can be managed e.g. by Scala Steward. We probably could source all dependencies from a `build_deps.sc` files, which we could also use in the outer test project, so Scala Steward can open pull requests. But this might result in a bit too complicated generated project. WDYT? Maybe, it also works, when we read the `default.properties` directly from the outer project. I'm currently not sure how exactly Scala Steward find matches.

Or we maintain the versions in two places, the properties file and a place where Scala Steward can update them. But we fail, when both are out of sync. Then, whenever we get a pull request, we need to manually update the properties file as well as part of merging them. Or we use some scala-fix setup, which can be automatically invoked by Scala Steward. But that all sounds rather complicated.

**Comment by ckipp01** [comment]
> Unfortunately, we loose some control over whether the resolved version combinations are guaranteed to work as expected.

Yea true. It's a tough call because it's a bummer to get a template that's out of date.

> Or we maintain the versions in two places, the properties file and a place where Scala Steward can update them. But we fail, when both are out of sync. Then, whenever we get a pull request, we need to manually update the properties file as well as part of merging them. Or we use some scala-fix setup, which can be automatically invoked by Scala Steward. But that all sounds rather complicated.

The [docs metion](https://github.com/scala-steward-org/scala-steward/blob/main/docs/faq.md#can-scala-steward-update-dependencies-in-giter8-templates-) adding the dependencies that you want Scala steward to update as part of your build. So in theory you should be able to include a `object Steward extends ScalaModule`, define the deps as `ivyDeps`, and then get them updated in your template as well. I haven't tried this yet with Mill but plan to. Would you want me to update this to take that approach and then add this to public Steward `repos.md` and then see if it works?

**Comment by lefou** [comment]
> The [docs metion](https://github.com/scala-steward-org/scala-steward/blob/main/docs/faq.md#can-scala-steward-update-dependencies-in-giter8-templates-) adding the dependencies that you want Scala steward to update as part of your build. So in theory you should be able to include a `object Steward extends ScalaModule`, define the deps as `ivyDeps`, and then get them updated in your template as well. I haven't tried this yet with Mill but plan to. Would you want me to update this to take that approach and then add this to public Steward `repos.md` and then see if it works?

Yes, that would be great.

File: src/main/g8/build.sc [hunk 1]
```diff
@@ -7,7 +7,7 @@ object $name;format="camel"$ extends ScalaModule {
 
   object test extends Tests with TestModule.Munit {
     def ivyDeps = Agg(
-      ivy"org.scalameta::munit::0.7.29"
+      ivy"org.scalameta::munit::0.7.28"
```
Comment (ckipp01) [inline src/main/g8/build.sc:10]: Purposefully made this older. I'll add this repo to Steward and then we can ensure this works.

**Comment by ckipp01** [comment]
So it detects munit, so this in theory should work:

```
❯ mill --import ivy:org.scala-steward:scala-steward-mill-plugin_2.13:0.15.0 show org.scalasteward.mill.plugin.StewardPlugin/extractDeps
Compiling /Users/ckipp/Documents/scala-workspace/mill-scala-hello.g8/build.sc
[1/1] show
[1/1] show > [6/6] org.scalasteward.mill.plugin.StewardPlugin.extractDeps
{
  "modules": [
    {
      "name": "Steward",
      "repositories": [
        {
          "url": "https://repo1.maven.org/maven2",
          "type": "maven",
          "auth": null
        },
      ],
      "dependencies": [
        {
          "groupId": "org.scalameta",
          "artifactId": {
            "name": "munit",
            "maybeCrossName": "munit_2.13"
          },
          "version": "0.7.28"
        }
      ]
    }
  ]
}
```

**Review by lefou — APPROVED** [review]
Looks good to me. Thank you!

**Pull request merged**